lib/src/main/java/growthbook/sdk/java/ExperimentEvaluator.java

```diff
@@ -126,49 +124,57 @@ public <ValueType> ExperimentResult<ValueType> evaluateExperiment(Experiment<Val
             if (filters != null) {
 
                 // Exclude if user is filtered out (used to be called "namespace")
-                if (GrowthBookUtils.isFilteredOut(filters, attributeOverrides, context)) {
-                    return getExperimentResult(context, experiment, -1, false, featureId, null, null, attributeOverrides);
+                if (GrowthBookUtils.isFilteredOut(filters, context.getUser().getAttributes())) {
+                    return getExperimentResult(context, experiment, -1, false, featureId, null, null);
                 }
             } else if (namespace != null) {
 
                 // If experiment namespace is set, check if the hash value is included in the range, and if not
                 // user is not in the experiment, variation 0.
                 Boolean isInNamespace = GrowthBookUtils.inNameSpace(hashAttribute.getHashValue(), namespace);
                 if (!isInNamespace) {
-                    return getExperimentResult(context, experiment, -1, false, featureId, null, null, attributeOverrides);
+                    return getExperimentResult(context, experiment, -1, false, featureId, null, null);
                 }
             }
 
             // Evaluate the condition JSON
             // can it be instead JsonObject
             JsonObject conditionJson = experiment.getConditionJson();
             if (conditionJson != null) {
-                Boolean shouldEvaluate = conditionEvaluator.evaluateCondition(attributes, conditionJson, context.getSavedGroups());
+                Boolean shouldEvaluate = conditionEvaluator.evaluateCondition(context.getUser().getAttributes(), conditionJson, context.getGlobal().getSavedGroups());
 
                 // If experiment.condition is set and the condition evaluates to false,
                 // return immediately (not in experiment, variationId 0)
                 if (!shouldEvaluate) {
-                    return getExperimentResult(context, experiment, -1, false, featureId, null, null, attributeOverrides);
+                    return getExperimentResult(context, experiment, -1, false, featureId, null, null);
                 }
             }
 
             // 10. Exclude if prerequisites are not met
             List<ParentCondition> parenConditions = experiment.getParentConditions();
             if (parenConditions != null) {
                 for (ParentCondition parentCondition : parenConditions) {
+
+                    UserContext user = UserContext.builder()
+                            .attributes(jsonUtils.gson.fromJson(parentCondition.getCondition(), JsonObject.class))
+                            .build();
```

Comment:
This doesn't feel right to me.  I'm pretty sure we don't want to modify the user's attributes or create a subContext at all here.  It looks like the original implementation was wrong here as well.